tests/ui_tools/test_boxes.py

```diff
@@ -103,6 +105,124 @@ def test_not_calling_typing_method_without_recipients(
+        )
+        write_box.msg_write_box = mocker.Mock(edit_text=draft_composition["content"])
+        write_box.stream_write_box = mocker.Mock(edit_text=draft_composition["to"])
+        write_box.title_write_box = mocker.Mock(edit_text=draft_composition["subject"])
+
+        write_box.compose_box_status = "open_with_stream"
+        write_box.stream_id = 1
+
+        size = widget_size(write_box)
+        write_box.keypress(size, key)
+
+        assert write_box.model.session_draft_message.called
+        if is_draft_saved_in_current_session:
+            write_box.view.controller.save_draft_confirmation_popup.assert_called_once_with(
+                draft_composition
+            )
+        else:
+            write_box.model.save_draft.assert_called_once_with(draft_composition)
+
+    @pytest.mark.parametrize("key", keys_for_command("SAVE_AS_DRAFT"))
+    @pytest.mark.parametrize(
+        "draft_composition",
+        [
+            case(
+                {
+                    "type": "private",
+                    "to": [5140],
+                    "content": "Random message",
+                    "read_by_sender": True,
+                },
+                id="dm",
+            ),
+            case(
+                {
+                    "type": "private",
+                    "to": [5140, 5179],
+                    "content": "Random huddle message",
+                    "read_by_sender": True,
+                },
+                id="huddle",
+            ),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "are_valid_recipients",
+        [True, False],
+        ids=["valid_recipients", "invalid_recipients"],
+    )
+    @pytest.mark.parametrize(
+        "is_draft_saved_in_current_session",
+        [True, False],
+        ids=["saved_draft_exists", "no_saved_draft_exists"],
+    )
+    def test_keypress_SAVE_DRAFT_MESSAGE_private(
+        self,
+        key: str,
+        mocker: MockerFixture,
+        draft_composition: PrivateComposition,
+        are_valid_recipients: bool,
+        is_draft_saved_in_current_session: bool,
+        write_box: WriteBox,
+        widget_size: Callable[[Widget], urwid_Size],
+    ) -> None:
+        mocker.patch(MODULE + ".WriteBox.update_recipients")
+        write_box.msg_write_box = mocker.Mock(edit_text=draft_composition["content"])
+        write_box.to_write_box = mocker.Mock()
+        mocker.patch(
+            MODULE + ".WriteBox._tidy_valid_recipients_and_notify_invalid_ones",
+            return_value=are_valid_recipients,
+        )
+
+        write_box.model.session_draft_message.return_value = (
+            is_draft_saved_in_current_session
+        )
+        write_box.compose_box_status = "open_with_private"
+        write_box.recipient_user_ids = draft_composition["to"]
+
+        size = widget_size(write_box)
+        return_value = write_box.keypress(size, key)
+
+        if not are_valid_recipients:
+            assert key == return_value
```

Comment:
For my understanding: why are we testing the key here? How is that dependent on the validity of the recipients? It seems to me (from a cursory glance) that `keypress` always returns its `key` argument.